pandas/core/indexes/datetimelike.py

```diff
@@ -206,7 +206,7 @@ def take(self, indices, axis=0, allow_fill=True, fill_value=None, **kwargs):
             self, indices, axis, allow_fill, fill_value, **kwargs
         )
 
-    @Appender(_shared_docs["searchsorted"])
+    @doc(IndexOpsMixin.searchsorted, klass="Datetime-like Index")
```

Comment:
Same comment - this is very confusing

Reply:
Hi @WillAyd, I agree with you and @ simonjayhawkins here. It might be confusing. However, the original docstring is not extended from the base class. It seems like the original code obscures the problem because it does not explicitly indicate the source of the docstring. I try to keep it as it is but use `@doc`.

It looks like we all agree this docstring template will confuse other developers, but do you feel needed to fix this issue in this PR? If so, what will be your suggestion? One option that comes in my mind will be using the docstring from the base class, and modify them to fit in this case. I was trying to avoid that because it will change the original docstring relations, and I am not sure if we did this on purpose. 

Just to be clarified, I am very willing to make the additional change to solve this confusion. I just don't know what will be the best way of doing that. One more thing, I have a [comment](https://github.com/pandas-dev/pandas/pull/31970#issuecomment-586631930) related to this. You might also be interested.

Reply:
Is there a more suitable location for the docstring then? Importing the IndexOpsMixin here is strange

Reply:
Personally I think this should be addressed in a follow up. It's here in `IndexOpsMixin` because that's where the `_shared_docs` was. I think this PR is already too complex to make the change here. And if we change `_shared_docs` before merging this, the conflict here will be quite annoying to fix. Does it make sense?

Reply:
> Is there a more suitable location for the docstring then? Importing the IndexOpsMixin here is strange

For me, I feel this situation might be caused by they are sharing the same interface, but don't have a common ancestor. If we have an interface declaration, I would say putting docstring there would be a good choice. However, Python as a dynamic programming language, we don't have to declare the interface.

Now, I don't have a good solution in mind, but I would like to look for it and see if we can find somewhere that makes more sense.